features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/crypto/historyvisible/HistoryVisibleStatePresenter.kt

```diff
@@ -35,11 +34,12 @@ class HistoryVisibleStatePresenter(
         val roomInfo by room.roomInfoFlow.collectAsState()
         // Implicitly assume the alert is initially acknowledged to avoid flashes in UI.
         val acknowledged by repository.hasAcknowledged(room.roomId).collectAsState(initial = true)
+        val isHistoryVisible = roomInfo.historyVisibility == RoomHistoryVisibility.Shared || roomInfo.historyVisibility == RoomHistoryVisibility.WorldReadable
 
         val coroutineScope = rememberCoroutineScope()
 
-        LaunchedEffect(roomInfo.historyVisibility, acknowledged) {
-            if (roomInfo.historyVisibility == RoomHistoryVisibility.Joined && acknowledged) {
+        LaunchedEffect(isHistoryVisible, acknowledged) {
+            if (!isHistoryVisible && acknowledged) {
```

Comment:
I'm not sure to follow why we have this here? Are we going to display the warning each time the historyVisibility comes back to `visible`?